gsl/span

```diff
@@ -359,6 +359,7 @@ public:
     constexpr static const index_type extent = Extent;
 
     // [span.cons], span constructors, copy, assignment, and destructor
+    template <bool Unused = false, class = std::enable_if_t<(Unused || Extent <= 0)>>
```

Comment:
Can you leave a comment that explains why Unused exists? I'll have to admit that I don't know why ;-)